x-pack/platform/plugins/shared/security/server/uiam/uiam_service.ts

```diff
@@ -47,6 +88,24 @@ export interface UiamServicePublic {
    * @param refreshToken UIAM session refresh token.
    */
   invalidateSessionTokens(accessToken: string, refreshToken: string): Promise<void>;
+
+  /**
+   * Grants an API key using the UIAM service.
+   * @param authorization The HTTP authorization header containing scheme and credentials.
+   * @param params The parameters for creating the API key (name and optional expiration).
+   * @returns A promise that resolves to an object containing the API key details.
+   */
+  grantApiKey(
+    authorization: HTTPAuthorizationHeader,
```

Comment:
From: https://github.com/elastic/kibana/pull/241050#discussion_r2549731959

> question/nit: how about accepting authorization: HTTPAuthorizationHeader instead of two separate args?

@azasypkin Agreed!